controller/CosmeticAPIController.py

```diff
@@ -0,0 +1,254 @@
+from sqlalchemy import exc
+from controller import app
+from dao import ProductRepo, BrandRepo, CategoryRepo, ColorRepo, ProductTypeRepo, TagRepo
+from dao.models import Product, Brand, Category, Color, ProductType, Tag
+
+
+@app.route("/")
+@jwt_required
+def home():
+    return app.send_static_file('index.html')
+
+
+"""
+API Call to get the product 
+Input : product_id
+"""
+
+
+@app.route('/product/<int:product_id>', methods=['GET'])
+@jwt_required
+def get_product(product_id):
+    try:
+        logging.info(f'getProduct: User has given input product_id as: {product_id}')
+        product = ProductRepo.getProduct(product_id)
+        if product is None:
+            return make_response(jsonify(response=f'Product with id {product_id} not found'), 404)
+        return jsonify(product=Product.serialize(product))
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
+
+
+"""
+API Call to search the product 
+Input: 
+        brand
+        product_category
+        product_type
+        price : greater_than 
+                less_than
+        rating:greater_than
+               less_than
+        tag : String or list separated by commas
+"""
+
+
+@app.route('/searchproduct', methods=['GET'])
+@jwt_required
+def search_product():
+    try:
+        products = ProductRepo.get_all_products(filters=request.args)
+        if len(products) > 0:
+            return jsonify(products=Product.serialize_list(products))
+        else:
+            return make_response(jsonify(response='No Products available for the given search criteria'), 200)
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
+
+
+"""
+Rest API : for add product
+Input: Allows to pass the new product details as
+                    request args or request body
+"""
+
+
+@app.route('/addproduct', methods=['POST'])
+@jwt_required
+def add_product():
+    try:
+        if len(request.args) == 0:
+            product = request.json
+        else:
+            product = json.loads(json.dumps(request.args))
+        if product is not None:
+            product_id = ProductRepo.create_product(product)
+        else:
+            return make_response(jsonify(response=f'Request data is invalid.'), 400)
+        return make_response(jsonify(response=f'Product added successfully and the product_id is {product_id} '), 201)
+    except sqlalchemy.exc.InvalidRequestError as reqerr:
+        return make_response(jsonify(error_message=str(reqerr)), 400)
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
```

Comment:
Python allows you to define except criteria for multiple exceptions:
```suggestion
    except (sqlalchemy.exc.InvalidRequestError, exc.SQLAlchemyError, Exception) as err:
        return make_response(jsonify(error_message=str(err)), 400)
```

It is not a good practice to catch a generic Exception and throw a 400 to the user. Many exceptions can be due to an internal error, which fall under 5XX segment. The information from the exception could be sensitive and reveal unwanted information to the user.